### PR TITLE
Adds TypeScript and JavaScript tabs to getting started theming screen

### PR DESCRIPTION
**Pull request by interdigitize**
*Issue #, if available:*

*Description of changes:*
Adds TypeScript and JavaScript tabs to getting started theming screen
Adds a component variation example to the "use the theme" step
Makes small formatting adjustments

Addresses comments from https://github.com/aws-amplify/amplify-ui/pull/960. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/src/pages/theming/getting-started.react.mdx [hunk 1]
```diff
@@ -1,48 +1,106 @@
...
 
-const MyComponent = ({ children }) => {
-  const { tokens } = useTheme();
-  return <Text color={tokens.colors.font.tertiary}>{children}</Text>;
-};
-```
+<Tabs>
+  <TabItem title="JavaScript">
+    ```jsx
+    // Option 1: Reference the theme object through component variations
```
Comment (reesscot) [inline docs/src/pages/theming/getting-started.react.mdx:19]:
```suggestion
    // Option 1: Use the theme through component variations
```

**Review by reesscot — COMMENTED** [review]
Thank you so much for your contribution @interdigitize! Just left a few change suggestions below.

File: docs/src/pages/theming/getting-started.react.mdx [hunk 2]
```diff
@@ -1,48 +1,106 @@
...
-import { Text, useTheme } from '@aws-amplify/ui-react';
+Step 2: Use the theme to style components
 
-const MyComponent = ({ children }) => {
-  const { tokens } = useTheme();
-  return <Text color={tokens.colors.font.tertiary}>{children}</Text>;
-};
-```
+<Tabs>
+  <TabItem title="JavaScript">
```
[the remark this answers is not in the document]
Comment (interdigitize) [reply docs/src/pages/theming/getting-started.react.mdx:17]: Yes, that makes sense. I'll adjust that.

File: docs/src/pages/theming/getting-started.react.mdx [hunk 3]
```diff
@@ -1,48 +1,106 @@
...
-    colors: {
-      font: {
-        primary: { value: '#008080' },
-        // ...
+  </TabItem>
+  <TabItem title="TypeScript">
+    ```jsx
+    // Option 1: Reference the theme object through component variations
+    import { Text } from '@aws-amplify/ui-react';
+    const MyComponent = ({ children }: React.PropsWithChildren<{}>) => {
```
[the remark this answers is not in the document]
Comment (interdigitize) [reply docs/src/pages/theming/getting-started.react.mdx:38]: Removed it.

File: docs/src/pages/theming/getting-started.react.mdx [hunk 4]
```diff
@@ -1,48 +1,106 @@
...
+    // Step 2: Pass the new theme to `AmplifyProvider`
+    // this will apply the theme to all Amplify UI components
+    <AmplifyProvider theme={theme}>
+      <App />
+    </AmplifyProvider>;
+    ```
+
+  </TabItem>
+  <TabItem title="TypeScript">
+    ```jsx
```
[the remark this answers is not in the document]
Comment (interdigitize) [reply docs/src/pages/theming/getting-started.react.mdx:82]: aha, nice. Updated.

**Comment by interdigitize** [comment]
@reesscot, I accepted/made the changes. 😊

I made one additional adjustment in the Optional step 2 example to align with the example in Step 1 so that it is explicitly clear what needs to be adjusted. Let me know if that needs any tweaking.

**Review by reesscot — APPROVED** [review]
LGTM. Thanks so much for your contribution! :shipit:

**Pull request merged**